src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,108 @@
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const initialNewMovieInfo = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  const [newMovie, setNewMovie] = useState(initialNewMovieInfo);
+  const [count, setCount] = useState(0);
+
+  const reset = () => setNewMovie(initialNewMovieInfo);
+
+  const pattern =
+    // eslint-disable-next-line max-len
+    /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+  const isFormValid = () => {
+    return (
+      newMovie.title.trim() &&
+      newMovie.imdbUrl.trim() &&
+      newMovie.imgUrl.trim() &&
+      newMovie.imdbId.trim() &&
+      pattern.test(newMovie.imdbUrl) &&
+      pattern.test(newMovie.imgUrl)
+    );
+  };
+
+  const handleChange = (name: keyof Movie, value: string) => {
+    setNewMovie(prev => ({
+      ...prev,
+      [name]: value,
+    }));
+  };
+
+  const onSubmitHandler = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!isFormValid()) {
+      return;
+    }
+
+    onAdd(newMovie);
+    setCount(count + 1);
+    reset();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={onSubmitHandler}>
       <h2 className="title">Add a movie</h2>
-
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={newMovie.title}
+        onChange={value => handleChange('title', value)}
         required
       />
 
-      <TextField name="description" label="Description" value="" />
+      <TextField
+        name="description"
+        label="Description"
+        value={newMovie.description}
+        onChange={value => handleChange('description', value)}
+      />
 
-      <TextField name="imgUrl" label="Image URL" value="" />
+      <TextField
+        name="imgUrl"
+        label="Image URL"
+        value={newMovie.imgUrl}
+        onChange={value => handleChange('imgUrl', value)}
+        required
+      />
 
-      <TextField name="imdbUrl" label="Imdb URL" value="" />
+      <TextField
+        name="imdbUrl"
+        label="Imdb URL"
+        value={newMovie.imdbUrl}
+        onChange={value => handleChange('imdbUrl', value)}
+        required
+      />
 
-      <TextField name="imdbId" label="Imdb ID" value="" />
+      <TextField
+        name="imdbId"
+        label="Imdb ID"
+        value={newMovie.imdbId}
+        onChange={value => handleChange('imdbId', value)}
+        required
+      />
 
       <div className="field is-grouped">
         <div className="control">
           <button
             type="submit"
             data-cy="submit-button"
             className="button is-link"
+            disabled={!isFormValid}
```

Comment:
```suggestion
            disabled={!isFormValid()}
```
it should fix failed test

Reply:
Great!
Thank you for your help!)